src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -2285,7 +2285,10 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         // If the very last thing in the list of recent commands, is exactly the
         // same as the current command, then let's not include it in the
         // history. It's literally the thing the user has typed, RIGHT now.
-        if (!commands.empty() && commands.back() == trimmedCurrentCommand)
+        // (also account for the fact that the cursor may be in the middle of a commandline)
+        if (!commands.empty() &&
+            !trimmedCurrentCommand.empty() &&
+            std::wstring{ commands.back() }.substr(0, trimmedCurrentCommand.size()) == trimmedCurrentCommand)

```

Comment:
Use `wstring_view` here. The commands vector isn't being deleted during this line after all.